tests/reflink.rs

```diff
@@ -57,11 +76,31 @@ fn reflink_src_is_dir() {
         Ok(()) => panic!(),
         Err(e) => {
             println!("{:?}", e);
-            assert_eq!(e.kind(), io::ErrorKind::InvalidInput)
+            // `io::ErrorKind::IsADirectory` is unstable
+            assert_eq!(e.kind().to_string(), "is a directory")
         }
     }
 }
 
+#[test]
```

Comment:
should be macos/ios only

Reply:
I added a check that we show the correct error on both macos (target exists) and not macos (invalid input)